### PR TITLE
Native Create Issue

### LINKED ISSUE #288: Add "Create Issue" button to repo view
Once native issue creation is supported, we should replace the "Report a Bug" button in Settings with a native UI.

### LINKED ISSUE #350: Create issue on project
Version 1.10.0 (1878)
Device: iPhone 7 Plus (iOS 11.0) 
Should prob go in the more options actionsheet?

### PR DESCRIPTION
**Pull request by Sherlouk**
Closes #288 
Closes #350 

- Created view controller to natively create an issue supporting a title and body text
- Pass through the correct GithubClient to Settings VC (necessary for correct authentication)
- Updated "Report a bug" tool to use issue creator (currently using a test repo)
- On a successful creation, the user is segued to the issues view controller (replacing new issue, so they can't go back to the form!)

Example output: https://github.com/rnystromtest/Githawk-Playground/issues/2
Used API: https://developer.github.com/v3/issues/#create-an-issue

### CODE REVIEW

**Comment by Sherlouk** [comment]
- Added new issue button to all repositories that have issue enabled
- Added markdown/action bar (includes a preview button)
- Fixed issues on tablet so things appear in the correct panel

image

Still looks like 💩 - If anyone wants to give a hand making it look nice I would appreciate it 😅

**Comment by Sherlouk** [comment]
I'd say for an initial implementation I'm happy with it now - Obviously we can add new functionality down the line for adding labels, assignees etc (assuming you have push permissions) but I think this is good?

**Comment by BasThomas** [comment]
Regarding design: a few years back I did something similar regarding native reporting, and think that looked pretty nice. Obviously a bit different in comparison with creating an issue, but design wise you get the idea.

screen shot 2017-09-16 at 17 13 32

**Comment by Sherlouk** [comment]
@BasThomas You're more than welcome to branch off and try match that!

It looks a look more modular, and similar to Eureka just couldn't see a nice way of doing it without ballooning hence with the simpler scroll view

**Comment by rnystrom** [comment]
Will give this a go soon, def need to massage the design. I personally prefer the look of the stuff @BasThomas posted: grouped `UITableView` style, tho it might be overkill for title+body. I'm not sure yet the best way to incorporate the editing tools. Input accessory view is prob the best bet for this.

**Comment by Sherlouk** [comment]
Design has never been a hotspot of mine, and editing tools are definitely up for discussion (I basically just copied what we had already to minimise on work!)

This does look quite nice though: http://1writerapp.com/
In essence a custom keyboard

**Comment by BasThomas** [comment]
Probably also want to add this option for repos you’re browsing via search. Will create a separate issue.

**Comment by Sherlouk** [comment]
This is available via search and for report a bug - am I missing something @BasThomas 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by BasThomas** [comment]
I don’t get it. 😬 What are you referring to?

I am expecting, if you look at the screenshot below, to also have an option to “Open an Issue”

0edf0c28-0397-40a9-aff0-b9fb07c285bf

File: Classes/Repository/RepositoryViewController.swift [hunk 1]
```diff
@@ -48,6 +48,30 @@ PrimaryViewController {
         feed.viewDidLoad()
         feed.adapter.dataSource = self
         title = "\(repo.owner)/\(repo.name)"
+        
+        if repo.hasIssuesEnabled {
+            navigationItem.rightBarButtonItem = UIBarButtonItem(
```
Comment (Sherlouk) [inline Classes/Repository/RepositoryViewController.swift]: @BasThomas You mean like this code here 😉

I know I need to move it to the action sheet, but the logic is all here to have a "New Issue" button for the repo view controller - Just don't merge master into this branch 😂

**Comment by Sherlouk** [comment]
@rnystrom Just updated with changes in master so the "Create Issue" is now in the overflow rather than it's own button

In terms of design, https://github.com/xmartlabs/Eureka would give us a basic form with all the structure done for us -- Would this be a solution you're okay with?

**Comment by rnystrom** [comment]
Is that project gonna be overkill tho? Will we use it for more stuff in the future? Nervous about taking another dependency for a simple design.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Sherlouk** [comment]
Further down the line I'm sure the form will get more complicated with files/images, labels, assignees, etc

Other forms include add/edit projects, maybe even editing repos

I'll give it a quick go now without Eureka and see how it ends up

**Comment by Sherlouk** [comment]
@rnystrom 

image

You mentioned before that we need to come up with a better way of doing markdown so I left the buttons out of this design. Is this what you had in mind?

**Comment by Sherlouk** [comment]
@BasThomas ^^ also

**Comment by rnystrom** [comment]
Ya that looks pretty great, nice and simple. We should be able to add the markdown actions as the input accessory view, right? @Sherlouk can you confirm that's still possible here?

**Comment by Sherlouk** [comment]
It should still be possible yea (it’s just a text view!) would we still want all the options though because it already has all the copy, paste, look up etc I feel like that Ui is soon going to get super busy

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Sherlouk** [comment]
image

Added markdown controls ^^

@rnystrom @BasThomas

**Comment by rnystrom** [comment]
Yessss!

**Comment by Sherlouk** [comment]
@rnystrom We waiting on anything to merge this now? 😄

**Comment by BasThomas** [comment]
Nice!

**Comment by rnystrom** [comment]
@Sherlouk don’t think so! I’ll give the code a quick look, but I also don’t mind refactoring anything on my own. Will merge today!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Sherlouk** [comment]
@rnystrom Hit me up when ready to merge and I can deal with the conflicts

**Comment by rnystrom** [comment]
@Sherlouk give it a go and DM me on Twitter so I can merge ASAP to avoid more conflicts!

**Comment by Sherlouk** [comment]
@rnystrom Picking this up now, will be 5 minutes

**Comment by Sherlouk** [comment]
All good!

**Comment by weyert** [comment]
May I ask why you have chosen a `UITableView` for this and not a `UICollectionView` or let's go wild and a `UIStackView`? I am still trying to find out when to use what. I really like `UICollectionView` myself as it's more powerful with it custom layout's support.

**Comment by Sherlouk** [comment]
Originally it was a `UIScrollView` with a couple elements in, ultimately I would've been happy with a bit of auto layout magic but crucially we still needed to support all the screen sizes so scrolling was a must

In the end it came down to simplicity and "out of the box" design. A grouped `UITableView` gives us a consistent design not only across the app but also familiarity across iOS. Using static cells makes it super easy to implement and all values can be accessed without the need for a maze of delegates and hierarchy calls.

In this situation there was very little flexibility required, though when we do get the point of adding additional fields, such as labels, this may have to change.

**Pull request merged**